frontend/screens/Influencer/components/UserProfilePhoto.js

```diff
@@ -1,20 +1,101 @@
-import React from "react";
-import { StyleSheet, View, Text, TouchableOpacity } from "react-native";
+import React, { useState } from "react";
+import {
+  StyleSheet,
+  View,
+  Text,
+  TouchableOpacity,
+  Alert,
+  Platform,
+  ScrollView,
+} from "react-native";
 import { Image } from "expo-image";
+import * as ImagePicker from "expo-image-picker";
 import { UserProfilePhotoStyles } from "./UserProfilePhoto.scss";
+import { useAlert } from "../../../util/AlertContext";
 
-const UserProfilePhoto = ({route,navigation}) => {
+const UserProfilePhoto = ({ route, navigation }) => {
+  const [selectedImage, setSelectedImage] = useState(null);
   const social = route.params?.social;
+  const {showAlert} = useAlert()
   const follower = route.params?.follower;
   const price = route.params?.price;
-  const handleUploadPhoto = () => {
-    // Handle the upload photo action
-    console.log("Upload photo clicked");
+  const [photo, setPhoto] = useState(null);
+  const requestPermission = async () => {
+    if (Platform.OS !== "web") {
+      const { status } =
+        await ImagePicker.requestMediaLibraryPermissionsAsync();
+      if (status !== "granted") {
+        showAlert("Alert","Permission to access media library is required!");
+        return false;
+      }
+    }
+    return true;
   };
 
-  const handleTakePhoto = () => {
-    // Handle the take photo action
-    console.log("Take photo clicked");
+  const handleUploadPhoto = async () => {
+    const permission = await requestPermission();
+    if (!permission) return;
+
+    const result = await ImagePicker.launchImageLibraryAsync({
+      mediaTypes: ImagePicker.MediaTypeOptions.Images,
+      allowsEditing: true,
+      aspect: [1, 1],
+      quality: 1,
+    });
+
+    if (!result.canceled) {
```

Comment:
I can see [this](https://github.com/Kajol-Kumari/InfluMart/pull/84/files#diff-09de8dbfc3a8a679e029667bb3e6b4362a8d35328f88db7048c4d9477366e637R46) to [this](https://github.com/Kajol-Kumari/InfluMart/pull/84/files#diff-09de8dbfc3a8a679e029667bb3e6b4362a8d35328f88db7048c4d9477366e637R58) logic multiple times, can we expose a method in util folder and reuse instead of writing the same code again and again?